Source/EasyNetQ/Consumer/InternalConsumerFactory.cs

```diff
@@ -2,20 +2,29 @@
 
 namespace EasyNetQ.Consumer
 {
+    /// <inheritdoc />
     public interface IInternalConsumerFactory : IDisposable
     {
+        /// <summary>
+        ///     Creates consumer based on configuration 
+        /// </summary>
+        /// <param name="configuration"></param>
+        /// <returns></returns>
         IInternalConsumer CreateConsumer(ConsumerConfiguration configuration);
     }
 
     /// <inheritdoc />
     public class InternalConsumerFactory : IInternalConsumerFactory
     {
-        private readonly IPersistentConnection connection;
+        private readonly IConsumerConnection connection;
         private readonly IEventBus eventBus;
         private readonly IHandlerRunner handlerRunner;
 
+        /// <summary>
+        ///     Creates InternalConsumerFactory
+        /// </summary>
         public InternalConsumerFactory(
-            IPersistentConnection connection, IHandlerRunner handlerRunner, IEventBus eventBus
+            IConsumerConnection connection, IHandlerRunner handlerRunner, IEventBus eventBus
```

Comment:
The same is here. A consumer could hang because of flow control.

https://github.com/EasyNetQ/EasyNetQ/pull/1273/files#r710740185